src/hooks/use-chat-context.tsx

```diff
@@ -0,0 +1,183 @@
+	useContext,
+	useMemo,
+	useState,
+	useEffect,
+	useCallback,
+	ReactNode,
+} from 'react';
+import { DEFAULT_PHP_VERSION } from '../../vendor/wp-now/src/constants';
+import { getIpcApi } from '../lib/get-ipc-api';
+import { useCheckInstalledApps } from './use-check-installed-apps';
+import { useGetWpVersion } from './use-get-wp-version';
+import { useSiteDetails } from './use-site-details';
+import { useWindowListener } from './use-window-listener';
+
+export interface ChatContextType {
+	currentURL: string;
+	pluginList: string[];
+	themeList: string[];
+	numberOfSites: number;
+	themeName?: string;
+	wpVersion: string;
+	phpVersion: string;
+	isBlockTheme?: boolean;
+	os: string;
+	availableEditors: string[];
+	siteName?: string;
+}
+const ChatContext = createContext< ChatContextType >( {
+	currentURL: '',
+	pluginList: [] as string[],
+	themeList: [] as string[],
+	numberOfSites: 0,
+	themeName: '',
+	phpVersion: '',
+	isBlockTheme: false,
+	wpVersion: '',
+	availableEditors: [] as string[],
+	os: '',
+	siteName: '',
+} );
+
+interface ChatProviderProps {
+	children: ReactNode;
+}
+
+const parseWpCliOutput = ( stdout: string, defaultValue: string[] ): string[] => {
+	try {
+		const data = JSON.parse( stdout );
+		return data?.map( ( item: { name: string } ) => item.name ) || [];
+	} catch ( error ) {
+		console.error( error );
+	}
+	return defaultValue;
+};
+
+export const ChatProvider: React.FC< ChatProviderProps > = ( { children } ) => {
+	const [ initialLoad, setInitialLoad ] = useState< Record< string, boolean > >( {} );
+	const installedApps = useCheckInstalledApps();
+	const { data: sites, loadingSites, selectedSite } = useSiteDetails();
+	const wpVersion = useGetWpVersion( selectedSite || ( {} as SiteDetails ) );
+	const [ pluginsList, setPluginsList ] = useState< Record< string, string[] > >( {} );
+	const [ themesList, setThemesList ] = useState< Record< string, string[] > >( {} );
+	const numberOfSites = sites?.length || 0;
+	const sitePath = selectedSite?.path || '';
+	const sitePort = selectedSite?.port || '';
+	const siteThemeDetails = selectedSite?.themeDetails;
+
+	const availableEditors = Object.keys( installedApps ).filter( ( app ) => {
+		return installedApps[ app as keyof InstalledApps ];
+	} );
+
+	const fetchPluginList = useCallback( async ( path: string ) => {
+		const { stdout, stderr } = await getIpcApi().executeWPCLiInline( {
+			projectPath: path,
+			args: 'plugin list --format=json --status=active',
+		} );
+		if ( stderr ) {
+			return [];
+		}
+		return parseWpCliOutput( stdout, [] );
+	}, [] );
+
+	const fetchThemeList = useCallback( async ( path: string ) => {
+		const { stdout, stderr } = await getIpcApi().executeWPCLiInline( {
+			projectPath: path,
+			args: 'theme list --format=json',
+		} );
+		if ( stderr ) {
+			return [];
+		}
+		return parseWpCliOutput( stdout, [] );
+	}, [] );
+
+	useEffect( () => {
+		let isCurrent = true;
+		// Initial load. Prefetch all the plugins and themes for the sites.
```

Comment:
The comment references "sites" in plural, however, I understand based on the logic that this initial load will be done only for the currently selected site. Is this accurate?

Reply:
Removed the comment, all along. I don't think it is really necessary. I think its clear what this use effect does by reading the code.